csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -91,22 +93,27 @@ public DependencyManager(string srcDir, IDependencyOptions options, ILogger logg
                     progressMonitor.MissingNuGet();
                 }
 
-                var restoredProjects = RestoreSolutions(solutions);
+                var restoredProjects = RestoreSolutions(solutions, out var assets1);
                 var projects = allProjects.Except(restoredProjects);
-                RestoreProjects(projects);
+                RestoreProjects(projects, out var assets2);
+
+                var dependencies = Assets.GetCompilationDependencies(progressMonitor, assets1.Union(assets2));
+                existsNetCoreRefNugetPackage = dependencies.Any(dependency => dependency.StartsWith("microsoft.netcore.app.ref"));
+                existsNetFrameworkRefNugetPackage = dependencies.Any(dependency => dependency.StartsWith("microsoft.netframework.referenceassemblies"));
+
+                var paths = dependencies
+                    .Select(d => Path.Combine(packageDirectory.DirInfo.FullName, d))
+                    .ToList();
+
+                // TODO: Rename the dllDirNames var - it's not only dirs anymore.
+                dllDirNames.AddRange(paths);
```

Comment:
What happens to the nuget packages that are restored but no in the `dependencies` list? Can there be any such package? What happens with asp.net core DLLs currently?

Reply:
(1) We ignore the nuget packages that are not in the depencies list (if there are such any). I have not tried to check the diff of download vs referenced (not sure what information that will give us).
(2) At least for new .NET projects, if they are ASP.NET projects then the ASP.NET framework is referenced in the `frameworkReferences` list.
(3) I think that the `project.assets.json` file also list the transitive package dependencies - these are ignored. For compiling, we should only need to include the packages that are direct dependencies.

Reply:
(1) I don't know if this can happen. We'd need to experiment with this a bit. It feels somewhat wrong to ignore restored nuget packages. But it also feels wrong to find fetched nuget packages that are not in the project.asset.json. 
(2) 👍 
(3) I think we do need transitive dependencies. If someone adds a dependency to `Newtonsoft.Json.Bson`, I think we expect to also have `Newtonsoft.Json` in our references. I quickly checked this locally, and it looks like the project.asset.json already has the reference to `Newtonsoft.Json`. So we don't need to do anything for this.